tools/cache.py

```diff
@@ -83,9 +83,8 @@ def ensure(self):
 
   def erase(self):
     with self.lock():
-      if self.dirname.exists():
-        for f in os.listdir(self.dirname):
-          tempfiles.try_delete(Path(self.dirname, f))
+      # Delete everything except the lockfile itself
+      utils.delete_contents(self.dirname, exclude=[os.path.basename(self.filelock_name)])
```

Comment:
is the exclusion part new in this PR?

Reply:
Before the removal of this file would silently fail (because of the try/catch and ignoring all failures).     Now we need to explicitly ignore this file, because we don't ignore failures.